### DISCUSSION TITLE
Масштабирование кластера

### DISCUSSION THREAD

**Discussion by Muhtarhan27** [autobase-tech/autobase#201] [General]
TASK [patroni : Prepare PostgreSQL | make sure the postgresql log directory "/var/log/postgresql" exists] ***********************************************************************************************************************************
ok: [10.190.46.172]
ok: [10.190.46.173]
changed: [10.190.46.192]

TASK [patroni : Prepare PostgreSQL | make sure PostgreSQL data directory "/var/lib/postgresql/14/main" exists] ******************************************************************************************************************************
ok: [10.190.46.172]
ok: [10.190.46.173]
changed: [10.190.46.192]

TASK [patroni : Prepare PostgreSQL | check that data directory "/var/lib/postgresql/14/main" is not initialized] ****************************************************************************************************************************
ok: [10.190.46.172]
ok: [10.190.46.173]
ok: [10.190.46.192]

TASK [patroni : Prepare PostgreSQL | data directory check result] ***************************************************************************************************************************************************************************
fatal: [10.190.46.172]: FAILED! => {"changed": false, "msg": "Whoops! data directory /var/lib/postgresql/14/main is already initialized"}
fatal: [10.190.46.173]: FAILED! => {"changed": false, "msg": "Whoops! data directory /var/lib/postgresql/14/main is already initialized"}

NO MORE HOSTS LEFT **************************************************************************************************************************************************************************************************************************

PLAY RECAP **********************************************************************************************************************************************************************************************************************************
10.190.46.171              : ok=2    changed=0    unreachable=0    failed=0    skipped=23   rescued=0    ignored=0
10.190.46.172              : ok=50   changed=8    unreachable=0    failed=1    skipped=153  rescued=0    ignored=1
10.190.46.173              : ok=48   changed=6    unreachable=0    failed=1    skipped=152  rescued=0    ignored=1
10.190.46.192              : ok=57   changed=43   unreachable=0    failed=0    skipped=145  rescued=0    ignored=1

**Comment by vitabaks** [comment]
Whoops! data directory /var/lib/postgresql/14/main is already initialized

**Reply by vitabaks** [reply]
the server you are trying to add to the cluster already contains PGDATA

**Reply by vitabaks** [reply]
if it's not production data, you can just clean up the  `/var/lib/postgresql/14/main`  directory before adding the server to the cluster.

**Comment by vitabaks** [comment]
In addition to deploying new clusters, this playbook also support the deployment of cluster over already existing and running PostgreSQL. You can convert your basic PostgreSQL installation to a high availability cluster. Just specify the variable `postgresql_exists='true'` in the inventory file.

**Comment by Muhtarhan27** [comment]
Thanks a lot !!!

**Reply by Muhtarhan27** [reply]
If one node could not be repaired, how can you get rid of ansible

**Comment by Muhtarhan27** [comment]
If one node could not be repaired, how can you get rid of ansible

**Reply by vitabaks** [reply]
I don't understand your question :)

**Comment by Muhtarhan27** [comment]
If one node could not be repaired, how can it be removed from the cluster
